sample_files/main.py

```diff
@@ -59,22 +87,19 @@
         response.raise_for_status()
 
         logger.info("Starting chain execution...")
-        chain_results = chain.invoke({"urls": read_urls(urls)})
+        chain_results = llm.invoke({"urls": read_urls(urls)})  # Assuming llm.invoke is the correct call
         time.sleep(2)
         logger.info("Chain execution successful.")
 
         if not check_chain_interactions(chain_results, urls):
             raise RuntimeError("Selenium process did not execute properly")
 
-        return
-
     except (RequestException, ValueError, RuntimeError) as e:
         logger.error(f"Error occurred: {e}")
-        logging.error(f"Error occurred: {e}")
-
-    logger.info(f"Retrying... Attempt {attempt}/{max_attempts}")
-    time.sleep(3)
-    execute_chain(urls, max_attempts, attempt + 1)
+        logger.info(f"Retrying... Attempt {attempt}/{max_attempts}")
+        time.sleep(3)
+        execute_chain(urls, max_attempts, attempt + 1)
 
 if __name__ == '__main__':
+    urls = ["http://example.com"]  # Replace with actual URLs
     execute_chain(urls)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

LGTM!


<!-- This is an auto-generated comment by OSS CodeRabbit -->